src/Simulation/EntryPointDriver/Azure.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.CommandLine.Parsing;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Runtime;
+
+namespace Microsoft.Quantum.CsharpGeneration.EntryPointDriver
+{
+    /// <summary>
+    /// Provides entry point submission to Azure Quantum.
+    /// </summary>
+    internal static class Azure
+    {
+        /// <summary>
+        /// Submits the entry point to Azure Quantum.
+        /// </summary>
+        /// <param name="entryPoint">The entry point.</param>
+        /// <param name="parseResult">The command-line parsing result.</param>
+        /// <param name="settings">The submission settings.</param>
+        /// <typeparam name="TIn">The entry point's argument type.</typeparam>
+        /// <typeparam name="TOut">The entry point's return type.</typeparam>
+        internal static async Task<int> Submit<TIn, TOut>(
+            IEntryPoint<TIn, TOut> entryPoint, ParseResult parseResult, AzureSettings settings)
+        {
+            var machine = CreateMachine(settings);
+            if (machine is null)
+            {
+                DisplayUnknownTargetError(settings.Target);
+                return 1;
+            }
+
+            var job = await machine.SubmitAsync(entryPoint.Info, entryPoint.CreateArgument(parseResult));
+            switch (settings.Output)
+            {
+                case OutputFormat.FriendlyUri:
+                    Console.WriteLine("Job submitted. To track your job status and see the results use:");
+                    Console.WriteLine();
+                    // TODO: Show the friendly URI.
+                    Console.WriteLine(job.Id);
+                    break;
+                case OutputFormat.Id:
+                    Console.WriteLine(job.Id);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException($"Invalid output format '{settings.Output}'.");
+            }
+            return 0;
+        }
+
+        /// <summary>
+        /// Creates a quantum machine based on the Azure Quantum submission settings.
+        /// </summary>
+        /// <param name="settings">The Azure Quantum submission settings.</param>
+        /// <returns>A quantum machine.</returns>
+        private static IQuantumMachine? CreateMachine(AzureSettings settings)
+        {
+            if (!(settings.Target is null) && settings.Target.StartsWith("ionq."))
+            {
+                // TODO: Give the number of shots to the quantum machine.
```

Comment:
same question here

Reply:
The `IQuantumMachine` interface does not support specifying the number of shots when a job is submitted. @cesarzc was planning to update it.